test/Microsoft.AspNetCore.JsonPatch.Test/Dynamic/DynamicObjectAdapterTests.cs

```diff
@@ -0,0 +1,1215 @@
+
+            Assert.Equal(new List<int>() { 4, 5, 6 }, doc.IntegerList);
+        }
+
+        [Fact]
+        public void ReplaceInListInList()
+        {
+            dynamic doc = new DynamicTestObject();
+            doc.SimpleDTOList = new List<SimpleDTO>() {
+                new SimpleDTO() {
+                    IntegerList = new List<int>(){1,2,3}
+                }};
+
+            // create patch
+            JsonPatchDocument patchDoc = new JsonPatchDocument();
+            patchDoc.Replace("SimpleDTOList/0/IntegerList/0", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDoc);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(doc);
+
+            Assert.Equal(4, doc.SimpleDTOList[0].IntegerList[0]);
+        }
+
+        [Fact]
+        public void ReplaceInListInListAtEnd()
+        {
+            dynamic doc = new DynamicTestObject();
+            doc.SimpleDTOList = new List<SimpleDTO>() {
+                new SimpleDTO() {
+                    IntegerList = new List<int>(){1,2,3}
+                }};
+
+            // create patch
+            JsonPatchDocument patchDoc = new JsonPatchDocument();
+            patchDoc.Replace("SimpleDTOList/0/IntegerList/-", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDoc);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(doc);
+
+            Assert.Equal(4, doc.SimpleDTOList[0].IntegerList[2]);
+        }
+
+        [Fact]
+        public void ReplaceFullListFromEnumerable()
+        {
+            dynamic doc = new DynamicTestObject();
+            doc.IntegerList = new List<int>() { 1, 2, 3 };
+
+            // create patch
+            JsonPatchDocument patchDoc = new JsonPatchDocument();
+            patchDoc.Replace("IntegerList", new List<int>() { 4, 5, 6 });
+
+            var serialized = JsonConvert.SerializeObject(patchDoc);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(doc);
+
+            Assert.Equal(new List<int>() { 4, 5, 6 }, doc.IntegerList);
+        }
+
+        [Fact]
+        public void ReplaceFullListWithCollection()
+        {
+            dynamic doc = new DynamicTestObject();
+            doc.IntegerList = new List<int>() { 1, 2, 3 };
+
+            // create patch
+            JsonPatchDocument patchDoc = new JsonPatchDocument();
+            patchDoc.Replace("IntegerList", new Collection<int>() { 4, 5, 6 });
+
+            var serialized = JsonConvert.SerializeObject(patchDoc);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(doc);
+
+            Assert.Equal(new List<int>() { 4, 5, 6 }, doc.IntegerList);
+        }
+
+        [Fact]
+        public void ReplaceAtEndOfList()
```

Comment:
This is wayyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy overtesting for dynamic. Most of these really feel like integration tests.